tensorflow/core/util/mkl_util.h

```diff
@@ -1821,15 +1821,21 @@ class MklPrimitiveFactory {
   /// For those legacy device(w/o AVX512 and AVX2),
   /// MKL-DNN GEMM will be used.
   static inline bool IsLegacyPlatform() {
-    return (!port::TestCPUFeature(port::CPUFeature::AVX512F) &&
-            !port::TestCPUFeature(port::CPUFeature::AVX2));
+    static bool is_legacy_platform = [] {
+      return (!port::TestCPUFeature(port::CPUFeature::AVX512F) &&
+              !port::TestCPUFeature(port::CPUFeature::AVX2));
+    }();
```

Comment:
Nit: This doesn't need to be a lambda. It can also be constant.
```suggestion
    static const bool is_legacy_platform = 
        (!port::TestCPUFeature(port::CPUFeature::AVX512F) &&
         !port::TestCPUFeature(port::CPUFeature::AVX2));
```

Reply:
Thanks, will do code change (same for the other method) today.

Reply:
done